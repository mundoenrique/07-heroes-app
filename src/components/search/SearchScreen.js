export default function SearchScreen() {
	return (
		<>
			<h1>SearchScreen</h1>
		</>
	);
}
