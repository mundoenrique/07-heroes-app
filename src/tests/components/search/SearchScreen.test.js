import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import SearchScreen from '../../../components/search/SearchScreen';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useNavigate: () => mockNavigate,
}));

describe('Pruebas en <SearchScreen />', () => {
	test('Debe de mostrar <SearchScreen /> correctamente', () => {
		const wrapper = mount(
			<MemoryRouter initialEntries={['/serach']}>
				<SearchScreen />
			</MemoryRouter>
		);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('.alert-info').text().trim()).toBe(
			'Realiza una busqueda'
		);
	});

	test('Debe de mostrar al superheroe enviado en queryString', () => {
		const wrapper = mount(
			<MemoryRouter initialEntries={['/serach?q=batman']}>
				<SearchScreen />
			</MemoryRouter>
		);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('input').prop('value')).toBe('batman');
	});

	test('Debe de mostrar -mensaje sin resultados', () => {
		const wrapper = mount(
			<MemoryRouter initialEntries={['/serach?q=jhgytfj']}>
				<SearchScreen />
			</MemoryRouter>
		);

		expect(wrapper.find('.alert-danger').text().trim()).toBe(
			'No se encontraron resultados para: (jhgytfj)'
		);
	});

	test('Debe de llamar al navigate al nuevo url', () => {
		const wrapper = mount(
			<MemoryRouter initialEntries={['/serach']}>
				<SearchScreen />
			</MemoryRouter>
		);

		wrapper.find('input').simulate('change', {
			target: {
				name: 'heroName',
				value: 'batman',
			},
		});

		wrapper.find('form').prop('onSubmit')({
			preventDefault: () => {},
		});

		expect(mockNavigate).toHaveBeenCalledWith('?q=batman');
	});
});
