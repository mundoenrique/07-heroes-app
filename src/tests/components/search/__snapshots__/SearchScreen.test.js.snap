// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchScreen /> Debe de mostrar <SearchScreen /> correctamente 1`] = `
Array [
  <h1>
    Busca un heroe
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Indica un heroe
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="heroName"
          onChange={[Function]}
          placeholder="Heroe"
          type="text"
          value=""
        />
        <button
          className="btn btn-primary mt-2"
        >
          Buscar...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        resultados
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Realiza una busqueda
      </div>
      <div
        className="col animate__animated animate__fadeInRight"
      />
    </div>
  </div>,
]
`;

exports[`Pruebas en <SearchScreen /> Debe de mostrar al superheroe enviado en queryString 1`] = `
Array [
  <h1>
    Busca un heroe
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Indica un heroe
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="heroName"
          onChange={[Function]}
          placeholder="Heroe"
          type="text"
          value="batman"
        />
        <button
          className="btn btn-primary mt-2"
        >
          Buscar...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        resultados
      </h4>
      <hr />
      <div
        className="col animate__animated animate__fadeInRight"
      >
        <div
          className="col"
        >
          <div
            className="card mb-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Batman"
                  className="img-fluid rounded-start"
                  src="/assets/images/dc-batman.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Batman
                  </h5>
                  <p
                    className="card-text"
                  >
                    Bruce Wayne
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Detective Comics #27
                    </small>
                  </p>
                  <a
                    className="btn btn-link"
                    href="/hero/dc-batman"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
